lib/framedivider.rb

```diff
@@ -5,94 +5,216 @@
+    msg << "Invalid polygon (overlaps?, 2nd argument) - skipping"
+    TBD.log(TBD::DEBUG, msg)
+    return false
+  end
+  if poly1.empty?
+    msg = ""
+    msg << "'#{id1}': " unless id1.empty?
+    msg << "Empty polygon (overlaps?, 1st argument) - skipping"
+    TBD.log(TBD::DEBUG, msg)
+    return false
+  end
+  if poly2.empty?
+    msg = ""
+    msg << "'#{id2}': " unless id2.empty?
+    msg << "Empty polygon (overlaps?, 2nd argument) - skipping"
+    TBD.log(TBD::DEBUG, msg)
+    return false
+  end
+
+  poly1.each do |pt|
+    unless pt.class == OpenStudio::Point3d
+      msg = "#{pt.class}? "
+      msg << "for '#{id1}': " unless id1.empty?
+      msg << "Expected OpenStudio 3D points (overlaps?, 1st argument - skipping"
+      TBD.log(TBD::DEBUG, msg)
+      return false
+    end
+  end
+  poly2.each do |pt|
+    unless pt.class == OpenStudio::Point3d
+      msg = "#{pt.class}? "
+      msg << "for '#{id2}': " unless id2.empty?
+      msg << "Expected OpenStudio 3D points (overlaps?, 2nd argument - skipping"
+      TBD.log(TBD::DEBUG, msg)
+      return false
+    end
+  end
+
+  return true if fits?(poly1, poly2)
+  return true if fits?(poly2, poly1)
+
+  ft = OpenStudio::Transformation::alignFace(poly1).inverse
+  ft_poly1 = flatZ( (ft * poly1).reverse )
+  ft_poly2 = flatZ( (ft * poly2).reverse )
 
-  parent = parent.get
-  ft = OpenStudio::Transformation::alignFace(parent.vertices).inverse
-  ft_parent = (ft * parent.vertices).reverse
-  ft_parent = flatZ(ft_parent)
+  ft_model = OpenStudio::Model::Model.new
+  surface1 = OpenStudio::Model::Surface.new(ft_poly1, ft_model)
+  surface2 = OpenStudio::Model::Surface.new(ft_poly2, ft_model)
 
-  siblings = []
-  model.getSubSurfaces.each do |sub|
-    next if sub.nameString == id
-    next if sub.surface.empty?
-    next unless sub.surface.get.nameString == parent.nameString
-    siblings << flatZ( (ft * sub.vertices).reverse )
+  area1 = surface1.grossArea
+  unless area1 > TOL
+    msg = ""
+    msg << "'#{id1}': " unless id1.empty?
+    msg << "Invalid polygon area (overlaps?, 1st argument) - skipping"
+    TBD.log(TBD::DEBUG, msg)
+    return false
+  end
+  area2 = surface2.grossArea
+  unless area2 > TOL
+    msg = ""
+    msg << "'#{id2}': " unless id2.empty?
+    msg << "Invalid polygon area (overlaps?, 2nd argument) - skipping"
+    TBD.log(TBD::DEBUG, msg)
+    return false
   end
 
-  ft_pts = flatZ(ft * pts)
-  ft_pts.each do |ft_pt|
-    return false unless OpenStudio::pointInPolygon(ft_pt, ft_parent, TOL)
-    siblings.each do |sibling|
-      return false if OpenStudio::pointInPolygon(ft_pt, sibling, TOL)
-    end
+  union = OpenStudio::join(ft_poly1, ft_poly2, TOL)
+  return false if union.empty?
+  union = union.get
+  surface = OpenStudio::Model::Surface.new(union, ft_model)
+  area = surface.grossArea
+  return false if area < TOL
+  if area1 > area2
+    return true if area - area1 > TOL
+  else
+    return true if area - area2 > TOL
   end
```

Comment:
```suggestion
```

If the result of join is not empty return true

Reply:
Yes simpler (thanks). I also kept a `return false if area < TOL` for the moment (in case). Upcoming commit.